bindgen/src/main/scala/render/enumeration.scala

```diff
@@ -35,5 +35,11 @@ def enumeration(model: Def.Enum, line: Appender)(using
 
       line(lhs + " = " + rhs)
     }
+    line(s"extension (a: $opaqueType)")
+    nest {
+      line(s"def &(b: $opaqueType): $opaqueType = a & b")
+      line(s"def |(b: $opaqueType): $opaqueType = a | b")
```

Comment:
Both of those raise an interesting problem - if I have an enum like this:

```c
enum Bla {
  X=2,
  Y=4
};
```

then `X & Y` is not actually of type `Bla`, because `(X & Y) != Bla.X && (X& Y) != Bla.Y`
Same with `|`.

I feel like in this case the return value of `&` and `|` should be the underlying integer type.

```scala
def &(b: Bla | Int): Int
```

(I added union type to support things like `Bla.X & Bla.Y & Bla.Z`)

And in C libraries there's mostly two patterns I've seen:

```c
void doSomething(Bla mode);
void doSomethingWithFlags(int flags);
```

where `flags` is expected to be a `&` or `|` combination of individual `Bla` enum cases

Reply:
Wouah ... No idea, I will try to understand this week :D

Reply:
I think this is worth the discussion, but for now this retains the C behaviour:

```c
#import "stdio.h"

typedef enum {
  A = 1,
  B = 2
} Hello;

int main() {
  Hello x = A & B;
  Hello y = A | B;

  printf("%d\n", x == A);
  printf("%d\n", y == A);
}
```

Outputs 

```
0
0
```

and the program typechecks.

So for now I will merge and the behaviour can be changed before 0.1.0. Thanks again!